### PR TITLE
Fix compile warnings

### PR DESCRIPTION
**Pull request by andyLiuChina**
this pull request include several type of compile warnings

### CODE REVIEW

File: tests/decodeinput.cpp [hunk 1]
```diff
@@ -214,11 +218,11 @@ bool DecodeInputVPX::init()
         fprintf (stderr, "fail to read ivf header, quit\n");
         return false;
     }
-    if (header.tag != 'FIKD')
+    if (header.tag != YAMI_FOURCC('F', 'I', 'K', 'D'))
         return false;
-    if (header.fourcc == '08PV')
+    if (header.fourcc == YAMI_FOURCC('0', '8', 'P', 'V'))
```
Comment (xuguangxin) [inline tests/decodeinput.cpp]: it need be YAMI_FOURCC('V', 'P', '8', '0')

File: tests/decodeinput.cpp [hunk 2]
```diff
@@ -33,6 +33,10 @@
 #include "decodeinputavformat.h"
 #endif
 
+#define YAMI_FOURCC(ch0, ch1, ch2, ch3) \
```
Comment (xuguangxin) [inline tests/decodeinput.cpp]: better put it to a command def.

File: tests/encodeinput.cpp [hunk 3]
```diff
@@ -35,6 +35,10 @@
 #include "encodeinput.h"
 #include "encodeInputDecoder.h"
 
+#define YAMI_FOURCC(ch0, ch1, ch2, ch3) \
```
Comment (xuguangxin) [inline tests/encodeinput.cpp]: do not repeat define here, just define in a command header

Comment (xuguangxin) [inline tests/encodeinput.cpp]: do not redefine it here just define it in a common header.

**Comment by YuJiankang** [comment]
+1

**Comment by xuguangxin** [comment]
merged as 7712df299b606058f0a4d7e21cb3d967f15663f3

**Pull request closed without merging**